packages/core/src/test/shared/errors.test.ts

```diff
@@ -467,21 +484,40 @@ describe('util', function () {
         // Arg withCause=true
         awsErr = new TestAwsError('ValidationException', 'aws validation msg 1', new Date())
         let toolkitError = new ToolkitError('ToolkitError Message')
-        assert.deepStrictEqual(getErrorMsg(toolkitError, true), 'ToolkitError Message')
+        assert.deepStrictEqual(getErrorMsg(toolkitError, { withCause: true }), 'ToolkitError Message')
 
         toolkitError = new ToolkitError('ToolkitError Message', { cause: awsErr })
-        assert.deepStrictEqual(getErrorMsg(toolkitError, true), `ToolkitError Message | aws validation msg 1`)
+        assert.deepStrictEqual(
+            getErrorMsg(toolkitError, { withCause: true }),
+            `ToolkitError Message | aws validation msg 1`
+        )
 
         const nestedNestedToolkitError = new ToolkitError('C')
         const nestedToolkitError = new ToolkitError('B', { cause: nestedNestedToolkitError })
         toolkitError = new ToolkitError('A', { cause: nestedToolkitError })
-        assert.deepStrictEqual(getErrorMsg(toolkitError, true), `A | B | C`)
+        assert.deepStrictEqual(getErrorMsg(toolkitError, { withCause: true }), `A | B | C`)
+
+        // Arg withCause=true, withId=true
+        const cError = new Error('C')
+        cError.name = 'NameC'
+        const bError = new ToolkitError('B', { cause: cError, code: 'CodeB' })
+        const aError = new ToolkitError('A', { cause: bError, code: 'CodeA' })
+        assert.deepStrictEqual(getErrorMsg(aError, { withCause: true, withId: true }), `CodeA: A | CodeB: B | NameC: C`)
```

Comment:
It looks like the use-case here is specifically for the `withCause` use-case where the error chain is denormalized into a single `reasonDesc` field.

So, there's no need to make `withId` optional. It should be controlled by the existing `withCause` field. It's not useful or wanted for any other case (we shouldn't be repeating `reason` in `reasonDesc`, normally).

Reply:
Good point, updating